src/core/qgscolorscheme.cpp

```diff
@@ -356,6 +356,12 @@ bool QgsUserColorScheme::erase()
     return false;
   }
 
+  // if file does not exist, nothing to do on the disk, so we can consider erasing done
+  if ( ! QFile::exists( filePath ) )
```

Comment:
How does this situation eventuate? Is the file deleted elsewhere in the code?

Reply:
Nope, seems like a "feature" of the module. Check https://github.com/suricactus/QGIS/blob/colorscheme_rm/src/core/qgscolorscheme.cpp#L319-L323 . I was afraid to `touch` the file there, as it might cause other implications.

Reply:
Maybe a little bit more context of the bug would be nice. You cannot delete a color scheme only if it is freshly created and you haven't add any colors to it. If you add at least one color, then it works fine.

Reply:
Thanks! That helps the review 😁